modules/currency-l0/src/main/scala/org/tessellation/currency/l0/CurrencyL0App.scala

```diff
@@ -58,8 +60,11 @@ abstract class CurrencyL0App(
 
     for {
       queues <- Queues.make[IO](sdkQueues).asResource
-      p2pClient = P2PClient.make[IO](sdkP2PClient, sdkResources.client, sdkServices.session)
       storages <- Storages.make[IO](sdkStorages, cfg.snapshot, method.globalL0Peer).asResource
+      selectedL0Peer <- storages.globalL0Cluster.getPeers.map { l0Peers =>
+        Random.shuffle(l0Peers.toNonEmptyList.toList).head
+      }.asResource
+      p2pClient = P2PClient.make[IO](sdkP2PClient, sdkResources.client, sdkServices.session, selectedL0Peer)
```

Comment:
as agreed offline, selected l0 peer should not be passed statically to `p2pClient` but should be selected just before fetching

Reply:
As discussed, `F[NonEmptySet[L0Peer]]` is passed to the `P2PClient`s and then `RandomPeerSelect`. That way, `RandomPeerSelect` has access to the latest discovered `L0Peer`s when select one of them.